query/src/main/kotlin/com/linecorp/kotlinjdsl/query/clause/from/Joiner.kt

```diff
@@ -49,23 +49,31 @@ class Joiner(
         realize(spec.entity) { query.from(spec.entity.type) }
     }
 
-    private fun realize(entity: EntitySpec<*>, join: () -> From<*, *>) {
+    private fun realize(entity: EntitySpec<*>, join: () -> Path<*>) {
         realized[entity] = join()
         realizedListeners.remove(entity)?.run {
             this.forEach { it() }
         }
     }
 
-    private fun join(spec: AssociatedJoinSpec<*, *>): From<*, *> = when (spec) {
-        is SimpleJoinSpec<*, *> -> realized.getValue(spec.left).join<Any, Any>(spec.path, spec.joinType)
-        is FetchJoinSpec<*, *> -> realized.getValue(spec.left).fetch(spec.path, spec.joinType)
-    } as Join<*, *>
+    private fun join(spec: AssociatedJoinSpec<*, *>): Path<*> = when (spec) {
```

Comment:
-- english
Since the left variable is commonly contained in the AssociatedJoinSpec, I think the alias part can be extracted.

-- korean
AssociatedJoinSpec에  left 변수가 공통으로 들어있기 때문에 alias 부분을 추출할 수 있을 것 같습니다.

```kotlin
spec
  .apply { spec.left.criteriaAlias()?.run { alias(this) } } 
  .let { when() }
```

Reply:
반영했습니다.